File: linked_list/0206_reverse_linked_list/0206_reverse_linked_list.go
package main

import "fmt"

type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
	var prev *ListNode

	for head != nil {
		temp := head.Next
		head.Next = prev
		prev = head
		head = temp
	}

	return prev
}

func main() {
	head := &ListNode{1, nil}
	head.Next = &ListNode{2, nil}
	head.Next.Next = &ListNode{3, nil}
	result := reverseList(head)
	fmt.Println(result.Next.Next.Val)
}

File: linked_list/0206_reverse_linked_list/0206_reverse_linked_list.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution(object):
    def reverseList(self, head):
        prev = None
        while head:
            curr = head
            head = head.next
            curr.next = prev
            prev = curr
        return prev


head = ListNode(1)
head.next = ListNode(2)
head.next.next = ListNode(3)
result = Solution().reverseList(head)
print(result.next.next.val)